## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 546306`
Commit: `17e57dcc14c7648fff00d2389d23f5f93fd423df`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python313Packages.connectrpc</li>
    <li>python313Packages.e2b</li>
    <li>python313Packages.e2b-code-interpreter</li>
    <li>python313Packages.protobuf-py</li>
    <li>python313Packages.protobuf-py-ext</li>
    <li>python313Packages.pyqwest</li>
    <li>python314Packages.connectrpc</li>
    <li>python314Packages.e2b</li>
    <li>python314Packages.e2b-code-interpreter</li>
    <li>python314Packages.protobuf-py</li>
    <li>python314Packages.protobuf-py-ext</li>
    <li>python314Packages.pyqwest</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python313Packages.connectrpc</li>
    <li>python313Packages.e2b</li>
    <li>python313Packages.e2b-code-interpreter</li>
    <li>python313Packages.protobuf-py</li>
    <li>python313Packages.protobuf-py-ext</li>
    <li>python313Packages.pyqwest</li>
    <li>python314Packages.connectrpc</li>
    <li>python314Packages.e2b</li>
    <li>python314Packages.e2b-code-interpreter</li>
    <li>python314Packages.protobuf-py</li>
    <li>python314Packages.protobuf-py-ext</li>
    <li>python314Packages.pyqwest</li>
  </ul>
</details>


---
### `aarch64-darwin`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python313Packages.e2b</li>
    <li>python313Packages.e2b-code-interpreter</li>
    <li>python314Packages.e2b</li>
    <li>python314Packages.e2b-code-interpreter</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python313Packages.connectrpc</li>
    <li>python313Packages.protobuf-py</li>
    <li>python313Packages.protobuf-py-ext</li>
    <li>python313Packages.pyqwest</li>
    <li>python314Packages.connectrpc</li>
    <li>python314Packages.protobuf-py</li>
    <li>python314Packages.protobuf-py-ext</li>
    <li>python314Packages.pyqwest</li>
  </ul>
</details>


---
<details>
<summary>python313Packages.e2b</summary>
<pre>FAILED tests/test_volume_client.py::test_async_explicit_stream_idle_timeout_above_transport_bound - PermissionError: [Errno 1] Operation not permitted
FAILED tests/test_volume_client.py::test_async_stream_idle_timeout_zero_disables_idle_bound - PermissionError: [Errno 1] Operation not permitted
FAILED tests/test_volume_client.py::test_sync_non_stream_read_survives_response_slower_than_idle_timeout - PermissionError: [Errno 1] Operation not permitted
FAILED tests/test_volume_client.py::test_sync_stream_response_head_is_bounded_by_idle_timeout - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_sync_transport_sends_proxy_credentials_and_headers - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_transport_emits_pyqwest_access_log - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_sync_api_client_round_trips_through_pyqwest - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_sync_api_client_serves_concurrent_threads - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_async_api_client_serves_concurrent_requests - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_async_api_client_round_trips_through_pyqwest - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_sync_api_client_leaves_redirects_to_httpx - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_async_api_client_leaves_redirects_to_httpx - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_sync_api_client_timeout_raises_httpx_read_timeout - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_async_api_client_timeout_raises_httpx_read_timeout - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_sync_api_client_body_timeout_raises_httpx_read_timeout - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_async_api_client_body_timeout_raises_httpx_read_timeout - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_sync_http1_transport_round_trips - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_async_http1_transport_round_trips - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_sync_transport_sends_multipart_bodies - PermissionError: [Errno 1] Operation not permitted
================== 34 failed, 207 passed, 15 errors in 1.47s ===================</pre>
</details>
<details>
<summary>python314Packages.e2b</summary>
<pre>FAILED tests/test_volume_client.py::test_async_explicit_stream_idle_timeout_above_transport_bound - PermissionError: [Errno 1] Operation not permitted
FAILED tests/test_volume_client.py::test_async_stream_idle_timeout_zero_disables_idle_bound - PermissionError: [Errno 1] Operation not permitted
FAILED tests/test_volume_client.py::test_sync_non_stream_read_survives_response_slower_than_idle_timeout - PermissionError: [Errno 1] Operation not permitted
FAILED tests/test_volume_client.py::test_sync_stream_response_head_is_bounded_by_idle_timeout - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_sync_transport_sends_proxy_credentials_and_headers - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_transport_emits_pyqwest_access_log - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_sync_api_client_round_trips_through_pyqwest - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_sync_api_client_serves_concurrent_threads - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_async_api_client_serves_concurrent_requests - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_async_api_client_round_trips_through_pyqwest - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_sync_api_client_leaves_redirects_to_httpx - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_async_api_client_leaves_redirects_to_httpx - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_sync_api_client_timeout_raises_httpx_read_timeout - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_async_api_client_timeout_raises_httpx_read_timeout - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_sync_api_client_body_timeout_raises_httpx_read_timeout - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_async_api_client_body_timeout_raises_httpx_read_timeout - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_sync_http1_transport_round_trips - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_async_http1_transport_round_trips - PermissionError: [Errno 1] Operation not permitted
ERROR tests/test_api_client_transport.py::test_sync_transport_sends_multipart_bodies - PermissionError: [Errno 1] Operation not permitted
================== 34 failed, 207 passed, 15 errors in 1.52s ===================</pre>
</details>

--- a/maintainers/maintainer-list.nix
+++ b/maintainers/maintainer-list.nix
@@ -18694,6 +18694,12 @@
     githubId = 54669781;
     keys = [ { fingerprint = "293B 93D8 A471 059F 85D7  16A6 5BA9 2099 D9BE 2DAA"; } ];
   };
+  mishushakov = {
+    email = "mish@e2b.dev";
+    github = "mishushakov";
+    githubId = 10400064;
+    name = "Mish Ushakov";
+  };
   misilelab = {
     name = "misilelab";
     email = "misileminecord@gmail.com";

--- a/pkgs/development/python-modules/protobuf-py-ext/default.nix
+++ b/pkgs/development/python-modules/protobuf-py-ext/default.nix
@@ -1,0 +1,49 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+
+  # build-system
+  rustPlatform,
+}:
+
+buildPythonPackage (finalAttrs: {
+  pname = "protobuf-py-ext";
+  version = "0.1.1";
+  pyproject = true;
+  __structuredAttrs = true;
+
+  # No tag on GitHub for this release: bufbuild/protobuf-py's only tag is v0.2.0.
+  # This has to stay in lockstep with protobuf-py, which is held at 0.1.1.
+  src = fetchPypi {
+    pname = "protobuf_py_ext";
+    inherit (finalAttrs) version;
+    hash = "sha256-6Fv9/bPtUGNNuMzHQp3ZKGUgEJSJxzVGOXGkGHB7T+8=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoVendor {
+    inherit (finalAttrs) pname version src;
+    hash = "sha256-oC9LPNUDVigOVqTHm3bRZlK0ToEHKlEpCtBRxqViFtg=";
+  };
+
+  nativeBuildInputs = [
+    rustPlatform.cargoSetupHook
+    rustPlatform.maturinBuildHook
+  ];
+
+  pythonImportsCheck = [ "protobuf_ext" ];
+
+  # The test suite lives in the protobuf-py repository and covers this extension
+  # only through protobuf-py's own tests, which the sdist does not ship.
+  # pythonImportsCheck confirms the extension loads for this interpreter, which
+  # is what packaging can get wrong.
+  doCheck = false;
+
+  meta = {
+    description = "Native accelerator for the protobuf-py runtime";
+    homepage = "https://github.com/bufbuild/protobuf-py";
+    # The sdist declares no license; the source repository is Apache-2.0.
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ mishushakov ];
+  };
+})

--- a/pkgs/development/python-modules/pyqwest/default.nix
+++ b/pkgs/development/python-modules/pyqwest/default.nix
@@ -1,0 +1,72 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+
+  # build-system
+  cmake,
+  rustPlatform,
+
+  # dependencies
+  opentelemetry-api,
+}:
+
+buildPythonPackage (finalAttrs: {
+  pname = "pyqwest";
+  version = "0.9.0";
+  pyproject = true;
+  __structuredAttrs = true;
+
+  src = fetchFromGitHub {
+    owner = "curioswitch";
+    repo = "pyqwest";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-K/zkPD7x3qKRPo7YJDWFrTLczaXvayKc8Zf/SQr2xL0=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoVendor {
+    inherit (finalAttrs) pname version src;
+    hash = "sha256-ztlUPsqEJz1WB2pXuoUyyhmUWY6MoWI/gErgg+6Fkcg=";
+  };
+
+  # reqwest' http3 feature refuses to compile without these cfgs, which upstream
+  # sets under `[build]`. Cargo ignores that section once cargoSetupHook has
+  # appended its own `[target.<host>]` rustflags, and setting RUSTFLAGS instead
+  # would shadow those. Flags from `cfg(...)` and `<triple>` sections are joined,
+  # so retargeting upstream's section keeps both sets.
+  postPatch = ''
+    substituteInPlace .cargo/config.toml \
+      --replace-fail '[build]' "[target.'cfg(all())']"
+  '';
+
+  nativeBuildInputs = [
+    # Only invoked by aws-lc-sys' build script; there is no top-level
+    # CMakeLists.txt to configure.
+    cmake
+    rustPlatform.bindgenHook
+    rustPlatform.cargoSetupHook
+    rustPlatform.maturinBuildHook
+  ];
+
+  dontUseCmakeConfigure = true;
+
+  dependencies = [
+    opentelemetry-api
+  ];
+
+  pythonImportsCheck = [ "pyqwest" ];
+
+  # The test suite spins up local HTTP/1, HTTP/2 and HTTP/3 servers through
+  # `pyvoy`, which is not packaged in nixpkgs. Nothing it covers is
+  # nixpkgs-specific: the packaging risk here is that the Rust extension fails
+  # to build or load, which pythonImportsCheck catches.
+  doCheck = false;
+
+  meta = {
+    description = "Modern, high-performance HTTP client for Python built on reqwest";
+    homepage = "https://github.com/curioswitch/pyqwest";
+    changelog = "https://github.com/curioswitch/pyqwest/releases/tag/${finalAttrs.src.tag}";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ mishushakov ];
+  };
+})

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -15557,6 +15557,8 @@ self: super: with self; {
 
   pyqvrpro = callPackage ../development/python-modules/pyqvrpro { };
 
+  pyqwest = callPackage ../development/python-modules/pyqwest { };
+
   pyqwikswitch = callPackage ../development/python-modules/pyqwikswitch { };
 
   pyrabbit2 = callPackage ../development/python-modules/pyrabbit2 { };

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -13663,6 +13663,8 @@ self: super: with self; {
   # If a protobuf upgrade causes many Python packages to fail, please pin it here to the previous version.
   protobuf = protobuf7;
 
+  protobuf-py-ext = callPackage ../development/python-modules/protobuf-py-ext { };
+
   protobuf3-to-dict = callPackage ../development/python-modules/protobuf3-to-dict { };
 
   # Protobuf 4.x

